examples/advanced_agents/k8s_infra_monitor/src/nat_k8s_infra_monitor/pod_health_tool.py

```
@@ -0,0 +1,148 @@
+    """Execute kubectl commands to gather live pod health data."""
+    cmd_base = ["kubectl"]
+    if kubeconfig_path:
+        cmd_base.extend(["--kubeconfig", kubeconfig_path])
+
+    ns_flag = ["--all-namespaces"] if not namespaces else []
+    sections: list[str] = []
+
+    if namespaces:
+        for ns in namespaces:
+            try:
+                result = subprocess.run(
```

Ruff (0.15.6)

[error] `subprocess` call: check for execution of untrusted input

(S603)